app/stac_api/serializers.py

```diff
@@ -802,10 +802,13 @@ class Meta:
     aborted = serializers.SerializerMethodField()
 
     def validate(self, attrs):
-        # TODO BGDIINF_SB-1983 md5_parts should be required not optional
+        # get partial from kwards (if partial true and no md5 : ok, if false no md5 : error)
         # Check the md5 parts length
-        if 'md5_parts' in attrs:
+        if attrs.get('md5_parts') is not None:
             validate_md5_parts(attrs['md5_parts'], attrs['number_parts'])
+        elif not self.partial:
+            raise ValidationError('')
```

Comment:
Here the error message should be set, for example `'md5_parts' argument missing`